### PR TITLE
Add support for generating and installing shell completions

### PR DESCRIPTION
**Pull request by Piccirello**
This supports generating completions for bash and zsh, and installing completions for bash.

Closes DPLR-585

### CODE REVIEW

**Review by rgmvisser — COMMENTED** [review]
Is it correct that you have to run a command to enable the completion? Can we have this on by default, or are there some downsides to that?

**Comment by Piccirello** [comment]
> Is it correct that you have to run a command to enable the completion? Can we have this on by default, or are there some downsides to that?

That is correct. I'm tracking automated installation in DPLR-1376.

File: pkg/cmd/completion.go [hunk 1]
```diff
@@ -16,35 +16,102 @@ limitations under the License.
...
+
+		filePath := fmt.Sprintf("%s/%s", path, name)
+		utils.LogDebug(fmt.Sprintf("Writing completion file to %s", filePath))
+		if err := utils.WriteFile(filePath, buf.Bytes(), 0644); err != nil {
+			utils.HandleError(err, "Unable to write completion file")
+		}
+
+		utils.Log("Your shell has been configured for Doppler CLI completions!")
+		utils.Log("")
+		utils.LogWarning("You must have the 'bash-completion' package installed for completions to work.")
```
Comment (ryan-blunden) [inline pkg/cmd/completion.go]:
```suggestion
		utils.Log("The homebrew 'bash-completion' package is required completions for to work. See https://docs.brew.sh/Shell-Completion for more info.")
```

File: pkg/cmd/completion.go [hunk 2]
```diff
@@ -16,35 +16,102 @@ limitations under the License.
...
+			}
 		}
+
+		filePath := fmt.Sprintf("%s/%s", path, name)
+		utils.LogDebug(fmt.Sprintf("Writing completion file to %s", filePath))
+		if err := utils.WriteFile(filePath, buf.Bytes(), 0644); err != nil {
+			utils.HandleError(err, "Unable to write completion file")
+		}
+
+		utils.Log("Your shell has been configured for Doppler CLI completions!")
```
Comment (ryan-blunden) [inline pkg/cmd/completion.go]:
```suggestion
		utils.Log("Your shell has been configured for Doppler CLI completions! Restart your shell to apply.")
```

**Review by rgmvisser — DISMISSED** [review]
Looks good from my side

**Review by ryan-blunden — DISMISSED** [review]
This is awesome! 

Consider updating the log messages to make it as obvious and easy as possible for auto-complete to start working, but not wedded to my suggestions.

**Comment by Piccirello** [comment]
Both great suggestions, will update.

**Comment by Piccirello** [comment]
@ryan-blunden I addressed your comments with some minor modifications. Can you take another look and lmk your thoughts ([diff](https://github.com/DopplerHQ/cli/compare/b696d3d38628a852872f32d12ee2be0c3be34ed9..a876ecd7c8ae4b4ec4c4844a7e99fbd28f773d8f))?

**Comment by ryan-blunden** [comment]
Brilliant! Helpful and to the point.

**Review by rgmvisser — APPROVED** [review]

**Pull request merged**